File: 66.cpp
// https://leetcode.com/problems/plus-one/submissions/
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int i = digits.size() - 1;
        int carry = 1;
        while (i >= 0) {
            digits[i] += carry;
            carry = 0;
            if (digits[i] == 10) {
                digits[i] = 0;
                carry = 1;
            }
            i--;
        }
        if (carry == 1) {
            digits.insert(digits.begin(), carry);
        }
        return digits;
    }
};

File: 66.py
# https://leetcode.com/problems/plus-one/
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        digits.reverse()
        carry = 1
        
        for i in range(len(digits)):
            digits[i] += carry
            if digits[i] == 10:
                carry = 1
                digits[i] = 0
            else:
                carry = 0
            
            if carry == 0:
                break
        
        if carry == 1:
            digits.append(1)
                
        digits.reverse()
        return digits
            